test/GitcoinGovernor.t.sol

```diff
@@ -29,3 +36,216 @@ contract GitcoinGovernorTest is Test, DeployInput {
+    jumpToActiveProposal();
+
+    // All delegates vote in support
+    for (uint _index = 0; _index < delegates.length; _index++) {
+      vm.prank(delegates[_index]);
+      governorAlpha.castVote(proposalId, true);
+    }
+
+    jumpToVoteComplete();
+  }
+
+  function passAndQueueProposal() public {
+    passProposal();
+    governorAlpha.queue(proposalId);
+  }
+
+  //--------------- TESTS ---------------//
+
+  function test_Proposal() public {
+    // Proposal has been recorded
+    assertEq(governorAlpha.proposalCount(), initialProposalCount + 1);
+
+    // Proposal is in the expected state
+    uint8 _state = governorAlpha.state(proposalId);
+    assertEq(_state, PENDING);
+
+    // Proposal actions correspond to Governor upgrade
+    (
+      address[] memory _targets,
+      uint256[] memory _values,
+      string[] memory _signatures,
+      bytes[] memory _calldatas
+    ) = governorAlpha.getActions(proposalId);
+    assertEq(_targets.length, 2);
+    assertEq(_targets[0], TIMELOCK);
+    assertEq(_targets[1], address(governor));
+    assertEq(_values.length, 2);
+    assertEq(_values[0], 0);
+    assertEq(_values[1], 0);
+    assertEq(_signatures.length, 2);
+    assertEq(_signatures[0], "setPendingAdmin(address)");
+    assertEq(_signatures[1], "__acceptAdmin()");
+    assertEq(_calldatas.length, 2);
+    assertEq(_calldatas[0], abi.encode(address(governor)));
+    assertEq(_calldatas[1], "");
+  }
+
+  function test_proposalActiveAfterDelay() public {
+    jumpToActiveProposal();
+
+    // Ensure proposal has become active the block after the voting delay
+    uint8 _state = governorAlpha.state(proposalId);
+    assertEq(_state, ACTIVE);
+  }
+
+  function testFuzz_ProposerCanCastVote(bool _willSupport) public {
+    jumpToActiveProposal();
+    uint256 _proposerVotes = gtcToken.getPriorVotes(PROPOSER, proposalStartBlock());
+
+    vm.prank(PROPOSER);
+    governorAlpha.castVote(proposalId, _willSupport);
+
+    IGovernorAlpha.Receipt memory _receipt = governorAlpha.getReceipt(proposalId, PROPOSER);
+    assertEq(_receipt.hasVoted, true);
+    assertEq(_receipt.support, _willSupport);
+    assertEq(_receipt.votes, _proposerVotes);
+  }
+
+  function test_ProposalSucceedsWhenAllDelegatesVoteFor() public {
+    jumpToActiveProposal();
+
+    // All delegates vote in support
+    for (uint _index = 0; _index < delegates.length; _index++) {
+      vm.prank(delegates[_index]);
+      governorAlpha.castVote(proposalId, true);
+    }
+
+    jumpToVoteComplete();
+
+    // Ensure proposal state is now succeeded
+    uint8 _state = governorAlpha.state(proposalId);
+    assertEq(_state, SUCCEEDED);
+  }
+
+  function test_ProposalDefeatedWhenAllDelegatesVoteAgainst() public {
+    jumpToActiveProposal();
+
+    // All delegates vote against
+    for (uint _index = 0; _index < delegates.length; _index++) {
+      vm.prank(delegates[_index]);
+      governorAlpha.castVote(proposalId, false);
+    }
+
+    jumpToVoteComplete();
```

Comment:
Same comment about `passProposal()`

Reply:
No, because in this case, we're not passing the proposal, we're defeating it.

Reply:
Ah right!